### PR TITLE
Matiasb/fix task refresh ical when empty value

### PR DESCRIPTION
**Pull request by matiasb**
This should fix task error as seen in logs, trying to parse an empty string as ical value:
```
Task apps.schedules.tasks.refresh_ical_files.refresh_ical_file[] raised unexpected: ValueError("Found no components where exactly one is required: ''")
```

### CODE REVIEW

File: engine/apps/schedules/tasks/refresh_ical_files.py [hunk 1]
```diff
@@ -41,22 +41,29 @@ def refresh_ical_file(schedule_pk):
         notify_ical_schedule_shift.apply_async((schedule.pk,))
 
     run_task_primary = False
-    if schedule.cached_ical_file_primary is not None:
-        if schedule.prev_ical_file_primary is None:
+    if bool(schedule.cached_ical_file_primary):
```
Comment (Konstantinov-Innokentii) [inline engine/apps/schedules/tasks/refresh_ical_files.py:44]: Why to convert to bool? Can't we check if schedule.cached_ical_file_primary, which will handle None end ""?

**Comment by Konstantinov-Innokentii** [comment]
@matiasb could you also check notify_ical_schedule_shift task for places with comparing ical to none? I'm sure there are some, there is one:

```
    for prev_ical_file, current_ical_file in prev_and_current_ical_files:
        if prev_ical_file is not None and (
            current_ical_file is None or not is_icals_equal(current_ical_file, prev_ical_file)
        ):
            # If icals are not equal then compare current_events from them
            is_prev_ical_diff = True
            prev_calendar = icalendar.Calendar.from_ical(prev_ical_file)

            prev_shifts_result, prev_users_result = get_current_shifts_from_ical(
                prev_calendar,
                schedule,
                prev_overrides_priority,
            )
            if prev_overrides_priority == 0 and prev_shifts_result:
                prev_overrides_priority = max([prev_shifts_result[uid]["priority"] for uid in prev_shifts_result]) + 1

            prev_shifts.update(prev_shifts_result)
            prev_users.update(prev_users_result)
```

File: engine/apps/schedules/tasks/refresh_ical_files.py (hunk 1 above)
Comment (matiasb) [reply engine/apps/schedules/tasks/refresh_ical_files.py]: I think it was more explicit, but switched back to just check for cached_ical_file_primary

**Comment by matiasb** [comment]
> @matiasb could you also check notify_ical_schedule_shift task for places with comparing ical to none? I'm sure there are some, there is one:

Checked and updated. Also did a search for any other occurrences of similar scenarios (nothing to add)

**Review by mderynck — APPROVED** [review]

**Pull request merged**